### PR TITLE
refactor: move components and tooltip to the s2-react from the s2-core

### LINKED ISSUE #243: core 层去除对 antd 的依赖
将  antd 改为 devDependencies, 在 __tests__ 之类的保证可以正常使用, 不作为生产依赖
- [ ] Tooltip
- [ ] 去掉 peerDependencies 的 react 声明
- [ ] rollup config external
- [ ] Menu

### PR DESCRIPTION
**Pull request by yangzhanmei**
### 👀 PR includes

✨ Feature

- [x] New feature

🎨 Enhance

- [x] Code style optimization
- [x] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [x] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
feat: move components and tooltip to s2-react from s2-core
feat: modify tooltip's default options
test: move sheetComponent's test to s2-react
### 🖼️ Screenshot

| Before                         | After                         |
| ------------------------------ | ------------------------------ |
| ❌                             | ✅                              |

### 🔗 Related issue link

close #243 

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

**Comment by lijinke666** [comment]
image

File: packages/s2-react/__tests__/data/tabular-data.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
+import { S2DataConfig, S2Options } from '@antv/s2';
+
+export const mockTabularOptions = {
+  width: 1000,
+  height: 600,
+  hierarchyType: 'grid',
+  style: {
```
Comment (xingwanying) [inline packages/s2-react/__tests__/data/tabular-data.ts:7]: 这个东西没了吧，感觉s2-react跟master有些滞后

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -34,12 +34,12 @@ jobs:
       - name: Install dependencies
         run: yarn
 
-      - name: Lint scripts, type, style and docs
-        run: yarn lint
-
       - name: Build
         run: yarn build
 
+      - name: Lint scripts, type, style and docs
```
Comment (lijinke666) [inline .github/workflows/lint.yml:40]: 这里换顺序是因为?

File: packages/s2-core/src/common/constant/options.ts [hunk 3]
```diff
@@ -39,12 +39,12 @@ export const DEFAULT_OPTIONS: Readonly<S2Options> = {
   conditions: {},
   totals: {},
   tooltip: {
-    showTooltip: true,
+    showTooltip: false,
```
Comment (lijinke666) [inline packages/s2-core/src/common/constant/options.ts:42]: 默认都不显示了吗?

File: packages/s2-core/src/common/icons/index.ts [hunk 4]
```diff
@@ -1,5 +1,4 @@
 import { getIcon, registerIcon } from './factory';
 import { GuiIcon } from './gui-icon';
-import { HtmlIcon } from './html-icon';
 
-export { GuiIcon, HtmlIcon, registerIcon, getIcon };
+export { GuiIcon, registerIcon, getIcon };
```
Comment (lijinke666) [inline packages/s2-core/src/common/icons/index.ts:4]:
```suggestion
export { getIcon, registerIcon } from './factory';
export { GuiIcon } from './gui-icon';
```

File: packages/s2-core/src/common/interface/tooltip.ts [hunk 5]
```diff
@@ -71,21 +69,21 @@ export type TooltipDetailProps = {
 
 export type TooltipInterpretationOptions = {
   name: string;
-  icon?: React.ReactNode;
+  icon?: Element | string;
```
Comment (lijinke666) [inline packages/s2-core/src/common/interface/tooltip.ts:72]: SVGElement | string ?

File: packages/s2-core/src/common/interface/tooltip.ts [hunk 6]
```diff
@@ -71,21 +69,21 @@ export type TooltipDetailProps = {
...
   text?: string;
-  render?: React.ElementType;
+  render?: Element | string;
 };
 
 export type InfosProps = {
-  infos: React.ReactNode;
+  infos: string;
```
Comment (lijinke666) [inline packages/s2-core/src/common/interface/tooltip.ts:78]: infos => info ?

File: packages/s2-core/src/ui/tooltip/index.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
+import { SpreadSheet } from '@/sheet-type';
```
Comment (lijinke666) [inline packages/s2-core/src/ui/tooltip/index.ts:1]: tooltip 加一点测试, 覆盖 public 的方法

File: packages/s2-react/README.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# S2-React
+
+## quick start
+
+```bash
+yarn install
+
+cd packages/s2-core && yarn build
+
+cd packages/s2-react && yarn start
```
Comment (lijinke666) [inline packages/s2-react/README.md:10]: yarn
yarn workspace @antv/s2-react build
yarn workspace @antv/s2-react start

另外启react包 还需要先 build core 层?

Comment (lijinke666) [reply packages/s2-react/README.md:10]: 集成到 根目录的 npm scripts

File: packages/s2-react/__tests__/data/data-custom-trees.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
+export const dataCustomTrees = [
+  {
+    'measure-a': 1,
+    type: '家具',
+    sub_type: '桌子',
```
Comment (lijinke666) [inline packages/s2-react/__tests__/data/data-custom-trees.ts:5]: 统一下命名风格 有横线, 又有下划线

File: packages/s2-react/__tests__/spreadsheet/filter-sheet-spec.tsx [hunk 10]
```diff
@@ -3,15 +3,15 @@ import React from 'react';
...
   S2DataConfig,
   S2Event,
   S2Options,
-  SheetComponent,
   SpreadSheet,
   TableSheet,
-} from '@/index';
+} from '@antv/s2';
+import { getContainer, getMockData } from '../util/helpers';
+import { SheetComponent } from '@/components/index';
```
Comment (lijinke666) [inline packages/s2-react/__tests__/spreadsheet/filter-sheet-spec.tsx:14]: @/components/index => @/components, 上同

File: packages/s2-core/src/common/constant/options.ts (hunk 3 above)
Comment (yangzhanmei) [reply packages/s2-core/src/common/constant/options.ts:42]: core层默认不显示了 s2-react默认显示

File: packages/s2-react/__tests__/data/tabular-data.ts (hunk 1 above)
Comment (yangzhanmei) [reply packages/s2-react/__tests__/data/tabular-data.ts:7]: 确实,合代码的时候合了半天,应该是漏改了,我删掉他

File: packages/s2-react/__tests__/data/data-custom-trees.ts (hunk 9 above)
Comment (yangzhanmei) [reply packages/s2-react/__tests__/data/data-custom-trees.ts:5]: 这个是迁移过来的, 暂时不动他了

File: packages/s2-core/src/common/interface/tooltip.ts (hunk 5 above)
Comment (yangzhanmei) [reply packages/s2-core/src/common/interface/tooltip.ts:72]: 用SVGElement会不会太限制了🤔

File: packages/s2-react/package.json [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+  "module": "esm/index.js",
+  "unpkg": "dist/index.js",
+  "types": "esm/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./esm/index.js",
+      "require": "./lib/index.js"
+    }
+  },
+  "description": "effective spreadsheet render lib",
```
Comment (lijinke666) [inline packages/s2-react/package.json:14]: 改下描述

File: packages/s2-react/package.json [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    "changelog": "conventional-changelog -p angular -i CHANGELOG.md -s"
+  },
+  "peerDependencies": {
+    "@antv/s2": "^0.2.5",
+    "react": "^16.14.0",
+    "react-dom": "^16.14.0"
+  },
+  "devDependencies": {
+    "@antv/s2": "^0.2.5",
+    "react": "^16.14.0",
```
Comment (lijinke666) [inline packages/s2-react/package.json:64]: 直接升 17?

File: packages/s2-core/src/common/icons/index.ts [hunk 13]
```diff
@@ -1,5 +1,2 @@
-import { getIcon, registerIcon } from './factory';
-import { GuiIcon } from './gui-icon';
-import { HtmlIcon } from './html-icon';
-
-export { GuiIcon, HtmlIcon, registerIcon, getIcon };
+export { getIcon, registerIcon } from './factory';
+export { GuiIcon } from './gui-icon';
```
Comment (qubaomingg) [inline packages/s2-core/src/common/icons/index.ts:2]: HtmlIcon 咋不用了？

Comment (yangzhanmei) [reply packages/s2-core/src/common/icons/index.ts:2]: 是react的组件 放到react里了

**Review by xingwanying — APPROVED** [review]
LG

**Pull request merged**